### ISSUE TITLE
Android: Modifying focus outline bounds

### ISSUE THREAD

**Issue by amarlette** [facebook/react-native#30852]
## Requires API Proposal
This issue requires a new API. An API proposal should be added and discussed before proceeding with implementation. The API proposal can be added in the comments of this issue or linked as a separate issue.

## Description
There is currently no way to modify the focus outline without also modifying the view's visible bounds.

On Android, this can be an important feature in some circumstances, as in older versions of Android (API 24 and lower) the way Talkback triggers a "click" is to fire a synthetic touch event at the center coordinates of this box.

## React Native version:
v0.63

## Expected Behavior
An API would need to be created that would allow setting the focus bounds. When this API was used, the focus bounds should be adjusted to match the values passed in.

## Android Details
This is a difficult issue to solve on Android, as there is currently no way to modify the focus bounds of a view.  You *can* however modify the focus bounds of a virtual view created via an Accessibility Delegate such as AndroidX's build in ExploreByTouchHelper. 

To solve this in RN you'd need a way to specify that the outputted View for a component should have an accessibility delegate attached that creates a virtual view identical to the original view, *except* with different bounds. This would require some deep diving into the system that translates JS components into native Views, and thorough testing to make sure that when creating a virtual view that all properties are correctly passed along.

## iOS Details

On iOS this is significantly easier than Android.  It would simply require modifying the accessibilityFrame property on the native View.

**Comment by blavalla** [comment]
I'm going to close this out as I don't think the complexity of implementing this on Android is worth the value provided by this feature.

Basically on Android this will require using virtual views rather than normal views, and supporting these for general purpose use would be quite complicated, as they would need to be attached to some non-virtual parent element, and parent elements can't mix children which are virtual and real, so using this on one element would require that all siblings are also virtual views.
